istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,564 @@
+	// Whether disable the webhook configuration of Sidecar Injector
+	disableInjectionWebhook bool
+	// The name of the injection webhook to manage
+	mutatingWebhookConfigName string
+}
+
+// Validate return nil if no error. Otherwise, return the error.
+func (opts *disableCliOptions) Validate() error {
+	if !opts.disableValidationWebhook && !opts.disableInjectionWebhook {
+		return fmt.Errorf("no webhook configuration to disable")
+	}
+	if opts.disableValidationWebhook && opts.validatingWebhookConfigName == "" {
+		return fmt.Errorf("validating webhook config to disable has empty name")
+	}
+	if opts.disableInjectionWebhook && opts.mutatingWebhookConfigName == "" {
+		return fmt.Errorf("mutating webhook config to disable has empty name")
+	}
+	return nil
+}
+
+type statusCliOptions struct {
+	// Whether display the webhook configuration of Galley
+	validationWebhook bool
+	// The name of the ValidatingWebhookConfiguration to manage
+	validatingWebhookConfigName string
+}
+
+// Webhook command to manage webhook configurations
+func Webhook() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "webhook",
+		Short: "webhook command to manage webhook configurations",
+	}
+
+	cmd.AddCommand(newEnableCmd())
+	cmd.AddCommand(newDisableCmd())
+	cmd.AddCommand(newStatusCmd())
+
+	return cmd
+}
+
+func newEnableCmd() *cobra.Command {
+	opts := &enableCliOptions{}
+	cmd := &cobra.Command{
+		Use:   "enable",
+		Short: "Enable webhook configurations",
+		Example: `
+# Enable the webhook configuration of Galley with the given webhook configuration
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --config-path validatingwebhookconfiguration.yaml
+
+# Enable the webhook configuration of Galley with the given webhook configuration and CA certificate
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --config-path validatingwebhookconfiguration.yaml --ca-bundle-file ./k8s-ca-cert.pem
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// --namespace is used as the namespace of the validation secret
+			opts.validationSecretNameSpace = namespace
+			if len(opts.webhookConfigPath) == 0 {
+				return fmt.Errorf("must specify a valid --config-path")
+			}
+			if len(opts.validationSecretName) == 0 {
+				return fmt.Errorf("must specify a valid --validation-secret")
+			}
+			if !opts.enableValidationWebhook {
+				return fmt.Errorf("not enabling validation webhook")
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating k8s client interface: %v", err)
+			}
+			// Read k8s CA certificate that will be used as the CA bundle of the webhook configuration
+			caCert, err := readCACert(client, opts.caCertPath, opts.validationSecretName, opts.validationSecretNameSpace,
+				opts.maxTimeForReadingWebhookCert)
+			if err != nil {
```

Comment:
Instead of saying `CA cert` use the same full name you use in StringVar()

Reply:
Done.